packages/rum/test/e2e/general-usecase/app.e2e-spec.js

```diff
@@ -129,4 +79,29 @@ describe('general-usercase', function() {
 
     return allowSomeBrowserErrors(['timeout test error with a secret'])
   })
+
+  it('should capture history.pushState', function() {
+    /**
+     * The query string is only used to make url different to the previous test,
+     * Otherwise, both tests will run in the same window.
+     */
+
+    browser.url('/test/e2e/general-usecase/index.html?run=pushState#test-state')
+    browser.waitUntil(
+      () => {
+        return $('#test-element').getText() === 'Passed'
+      },
+      5000,
+      'expected element #test-element'
+    )
+
+    const serverCalls = waitForApmServerCalls(0, 1)
+    expect(serverCalls.sendTransactions.length).toBe(1)
+    const transactionPayload = serverCalls.sendTransactions[0].args[0][0]
+    expect(transactionPayload.name).toBe('Push state title')
+    /**
+     * The actual spans are tested as part of the previous test.
+     */
+    expect(transactionPayload.spans.length).toBeGreaterThanOrEqual(3)
```

Comment:
Does this mean we will see the transaction in the UI as page-load transaction type and also see Resource and Nav timing spans as part of the History API route change?

Reply:
Yes, that's correct! I think it makes sense to look at the first route-change as an extension of the page-load therefore showing the continuity of the application.